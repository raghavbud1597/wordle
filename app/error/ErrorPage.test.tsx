import { render, screen } from "@testing-library/react";
import ErrorPage from "./page"; 
import { describe, it, expect } from "vitest";
import "@testing-library/jest-dom";

describe("ErrorPage", () => {
  it("renders an error image", () => {
    render(<ErrorPage />);
    const image = screen.getByAltText(/error image/i);
    expect(image).toBeInTheDocument();
  });

  it("displays the correct error message", () => {
    render(<ErrorPage />);
    const errorMessage = screen.getByText(/oops! something went wrong/i);
    expect(errorMessage).toBeInTheDocument();
  });

  it("renders a link to navigate back to the homepage", () => {
    render(<ErrorPage />);
    const link = screen.getByRole("link", { name: /back to the homepage/i });
    expect(link).toBeInTheDocument();
    expect(link).toHaveAttribute("href", "/");
  });

  it("matches the snapshot", () => {
    const { asFragment } = render(<ErrorPage />);
    expect(asFragment()).toMatchSnapshot();
  });
});
