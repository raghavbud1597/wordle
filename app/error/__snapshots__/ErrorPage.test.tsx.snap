// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ErrorPage > matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col justify-center items-center h-screen text-center"
  >
    <img
      alt="Error Image"
      class="mb-4"
      data-nimg="1"
      decoding="async"
      height="300"
      loading="lazy"
      src="/app/assets/error.svg"
      style="color: transparent;"
      width="300"
    />
    <h2
      class="text-2xl font-bold mb-4"
    >
      Oops! Something went wrong.
    </h2>
    <a
      aria-label="Back to the homepage"
      class="bg-white text-black border-2 border-black rounded-full hover:bg-black hover:text-white hover:border-white px-12 py-2 mx-4 text-lg"
      href="/"
    >
      Back to Home
    </a>
  </div>
</DocumentFragment>
`;
